### PR TITLE
Add ability to configure batch client retries

### PR DESCRIPTION
**Pull request by kankou-aliaksei**

### CODE REVIEW

File: miniwdl_aws.cfg [hunk 1]
```diff
@@ -24,6 +24,12 @@ download_concurrency = 5
 # configure AWS Batch jobs to mount the filesystem at this same location.
 root = /mnt/efs
 
+[batch_client]
+default_retries = {
```
Comment (mlin) [inline miniwdl_aws.cfg:28]: Please put this in the existing `[aws]` section, call it something like `boto3_retries` to disambiguate it from WDL-level task retries, and add a descriptive comment referring to the [relevant boto3 docs](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/retries.html#defining-a-retry-configuration-in-a-config-object-for-your-boto3-client).

File: miniwdl_aws/batch_job.py [hunk 2]
```diff
@@ -179,10 +180,14 @@ def _run(self, logger, terminating, command):
         Run task
         """
         try:
+            if os.environ.get("MINIWDL__AWS__BATCH_CLIENT_RETRIES") is not None:
+                aws_batch_client_retries = json.loads(os.environ["MINIWDL__AWS__BATCH_CLIENT_RETRIES"])
+            else:
+                aws_batch_client_retries = self.cfg["batch_client"].get_dict("default_retries")
```
Comment (mlin) [inline miniwdl_aws/batch_job.py:186]: The configuration loader (`self.cfg`) should honor the environment variable automatically, i.e. it shouldn't be necessary to explicitly check the environment variable above. [details](https://github.com/chanzuckerberg/miniwdl/blob/eedf2e04642f82b4337033dda21b1dce8eadcac0/WDL/runtime/config.py#L54-L72)

**Review by mlin — CHANGES_REQUESTED** [review]
Thanks for this @kankou-aliaksei 

High level comment. Since we currently enable the "standard" retry policy, I'd prefer to also address the root cause of whatever reliability issues we're still seeing -- not only cover it up with fancier retry logic. For example, if we're hitting request rate limits, then we should make sure the user can adjust relevant existing options (`submit_period`, `describe_period`). Relevant, https://github.com/aws/amazon-genomics-cli/issues/396

Other comments besides the inlines:

1. You should be able to `pre-commit run --all-files` to run the code formatter and linter locally to make the CI happy. We should have a CONTRIBUTING.md noting this.
2. I also need to change the CI to avoid trying to push to the docker registry from external PRs (which GitHub, properly, won't allow).

**Comment by mlin** [comment]
Update, I changed the GitHub workflows so it should avoid trying to touch GHCR from a pull request. If you rebase on main then it should run the linters only.

**Comment by kankou-aliaksei** [comment]
> Update, I changed the GitHub workflows so it should avoid trying to touch GHCR from a pull request. If you rebase on main then it should run the linters only.

Hi @mlin I've synchronized forked repo with original and merged main to feature branch however I still see ghcr checking.

**Comment by mlin** [comment]
Thanks @kankou-aliaksei , I made an additional change to how the configuration option is looked up, which is a workaround for a [problem in miniwdl](https://github.com/chanzuckerberg/miniwdl/issues/562) I need to fix separately. Sorry about that. Also further changed the CI to not run the GHCR stuff from forks.

**Pull request merged**